src/opendr/perception/object_detection_2d/nanodet/nanodet_learner.py

```diff
@@ -333,11 +360,147 @@ def reset(self):
         """This method is not used in this implementation."""
         return NotImplementedError
 
-    def optimize(self):
-        """This method is not used in this implementation."""
-        return NotImplementedError
+    def _save_onnx(self, onnx_path, img=None, do_constant_folding=False, verbose=True):
+        if self.jit_model:
+            print("Warning: A jit model has already initialized, inference will run in onnx mode by default!!!")
```

Comment:
This would be a good place to use warnings.warn.
```suggestion
            print("Warning: A JIT model was already initialized, inference will run in ONNX mode by default.")
```